openedx/core/djangoapps/credit/models.py

```diff
@@ -52,6 +85,46 @@ class Meta(object):
         """Model metadata"""
         unique_together = ('namespace', 'name', 'course')
 
+    @classmethod
+    def add_course_requirement(cls, credit_course, requirement):
+        """ Add requirements to given course
+
+    Args:
+        credit_course(CreditCourse): The identifier for credit course course
+        requirements(dict): Dict of requirements to be added
+
+    Raises:
+        InvalidCreditRequirements if any issue on requirement
+
+    Returns:
+        None
+    """
+        try:
+            cls.objects.create(
+                course=credit_course,
+                namespace=requirement["namespace"],
+                name=requirement["name"],
+                configuration=requirement["configuration"]
+            )
+        except IntegrityError:
+            # skip the duplicate entry
```

Comment:
@rlucioni The API function `set_credit_requirements()` is responsible for updating requirements (including modifying and disabling existing requirements).  I imagine that `set_credit_requirements()` will need to use `add_course_requirement` if the requirement doesn't already exist, but that we'll need additional code to modify and disable existing requirements.  Does that make sense?